scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala

```diff
@@ -185,7 +185,7 @@ final case class DisableSyntax(
   override def fix(ctx: RuleCtx): Patch = {
     val lints =
       (checkTree(ctx) ++ checkTokens(ctx) ++ checkRegex(ctx)).map(ctx.lint)
-    fixTree(ctx) ++ lints
+    (fixTree(ctx) ++ lints).atomic
```

Comment:
This would disable the rule for a full file if a single `// scalafix:ok` exists in the file. `.atomic` should be targeted towards a specific fix, for example on line 181 above